.github/workflows/reusable_build.yml

```diff
@@ -0,0 +1,50 @@
+name: Functional tests using Ragger
+
+on:
+  workflow_call:
+    inputs:
+      uploaded_artifact_name:
+        required: true
+        type: string
+
+jobs:
+  build:
+    name: Build application for NanoS, X and S+
+    strategy:
+      matrix:
+        include:
+          - SDK: "$NANOS_SDK"
+            model: nanos
+          - SDK: "$NANOX_SDK"
+            model: nanox
+          - SDK: "$NANOSP_SDK"
+            model: nanosp
+          - SDK: "$STAX_SDK"
+            model: stax
+
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-builder-lite:latest
+
+    steps:
+      - name: Clone
+        uses: actions/checkout@v3
+
+      - name: Build
+        run: |
+          git config --global --add safe.directory "$GITHUB_WORKSPACE"
+          make BOLOS_SDK=${{ matrix.SDK }}
+
+      - name: Get APPNAME value from Makefile
+        run: |
+          touch makefile_dumper.mk
+          echo "include Makefile" >> makefile_dumper.mk
+          echo "dump_appname:" >> makefile_dumper.mk
+          echo -e "\t@echo \$(APPNAME) | tr '[:upper:]' '[:lower:]'" >> makefile_dumper.mk
+          mv bin/app.elf "bin/$(make BOLOS_SDK=${{ matrix.SDK }} -f makefile_dumper.mk dump_appname | tail -n1)_${{ matrix.model }}.elf"
```

Comment:
I don't like this, but soon we could remove it once we don't use a separate elf folder for ragger I guess.

Reply:
This step has been removed following the separate build mechanism